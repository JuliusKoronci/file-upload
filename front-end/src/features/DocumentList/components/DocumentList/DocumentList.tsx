import { List } from '@material-ui/core';
import { SFC } from 'react';
import * as React from 'react';
import { IDocument } from '../../../../domains/Documents';

import { DocumentItem } from '../DocumentItem';


interface IDocumentListProps {
  items: IDocument[]
}

export const DocumentList: SFC<IDocumentListProps> = ({ items }) => (
  <List>
    {items.map(item => <DocumentItem item={item} key={item.slug} />)}
  </List>
);