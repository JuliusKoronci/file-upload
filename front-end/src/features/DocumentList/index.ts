/**
 * Public API of DocumentList
 */
export {
  DocumentListContainer as DocumentList,
}from './containers/DocumentListContainer';