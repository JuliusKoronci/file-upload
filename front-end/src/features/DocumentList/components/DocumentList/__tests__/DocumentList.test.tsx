import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { DocumentList } from '../DocumentList';


describe('DocumentList', () => {
  it('should render snapshot', () => {
    expect(toJson(shallow(<DocumentList
      documentState={{
        error: false,
        items: [{
          link: 'http://link',
          name: 'test',
          slug: 'test',
          type: 'jpg',
        }],
        loaded: true,
        loading: false,
      }}
    />))).toMatchSnapshot();
  });
  it('should render snapshot with empty list', () => {
    expect(toJson(shallow(<DocumentList
      documentState={{
        error: false,
        items: [],
        loaded: true,
        loading: false,
      }}
    />))).toMatchSnapshot();
  });
  it('should render snapshot with loading', () => {
    expect(toJson(shallow(<DocumentList
      documentState={{
        error: false,
        items: [],
        loaded: true,
        loading: true,
      }}
    />))).toMatchSnapshot();
  });
  it('should render snapshot with error', () => {
    expect(toJson(shallow(<DocumentList
      documentState={{
        error: 'some error',
        items: [],
        loaded: true,
        loading: false,
      }}
    />))).toMatchSnapshot();
  });
});