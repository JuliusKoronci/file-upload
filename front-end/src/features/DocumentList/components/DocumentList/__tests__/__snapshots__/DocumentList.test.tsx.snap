// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentList should render snapshot 1`] = `
<Component
  loading={false}
>
  <WithStyles(List)>
    <Component
      item={
        Object {
          "link": "http://link",
          "name": "test",
          "slug": "test",
          "type": "jpg",
        }
      }
      key="test"
    />
  </WithStyles(List)>
</Component>
`;

exports[`DocumentList should render snapshot with empty list 1`] = `
<Component
  loading={false}
>
  <WithStyles(List) />
</Component>
`;

exports[`DocumentList should render snapshot with error 1`] = `
<Component
  loading={false}
>
  <WithStyles(List) />
</Component>
`;

exports[`DocumentList should render snapshot with loading 1`] = `
<Component
  loading={true}
>
  <WithStyles(List) />
</Component>
`;
